django-backend/fecfiler/reports/management/commands/delete_committee_reports.py

```diff
@@ -0,0 +1,69 @@
+from django.core.management.base import BaseCommand
+from fecfiler.reports.models import Report
+from fecfiler.transactions.models import Transaction
+from fecfiler.memo_text.models import MemoText
+from fecfiler.committee_accounts.models import CommitteeAccount
+from fecfiler.web_services.models import DotFEC, UploadSubmission, WebPrintSubmission
+import re
+
+
+class Command(BaseCommand):
+    help = "Delete all reports (and transactions) for a given committee"
+
+    def add_arguments(self, parser):
+        parser.add_argument("committee_id", nargs=1, type=str)
+
+    def handle(self, *args, **options):
+        committee_ids = options.get('committee_id', None)
+        if committee_ids is None or len(committee_ids) == 0:
+            self.stderr.write("Error: no committee id provided")
+            return
+        committee_id = committee_ids[0]
+
+        cid_regex = re.compile("^C[0-9]{8}$")
+        if not cid_regex.match(str(committee_id)):
+            self.stderr.write(f'Error: invalid committee id "{committee_id}"')
+            return
+
+        committee = CommitteeAccount.objects.filter(committee_id=committee_id).first()
+        if committee is None:
+            self.stderr.write("Error: no matching committee")
+            return
+
+        self.delete_committee_reports(committee_id)
+
+    def delete_committee_reports(self, committee_id):
+        reports = Report.objects.filter(committee_account__committee_id=committee_id)
+        report_count = reports.count()
+        transactions = Transaction.objects.filter(
+            committee_account__committee_id=committee_id
+        )
+        transaction_count = transactions.count()
+        memo_count = MemoText.objects.filter(
+            report__committee_account__committee_id=committee_id
+        ).count()
+        dot_fec_count = DotFEC.objects.filter(
+            report__committee_account__committee_id=committee_id
+        ).count()
+        upload_submission_count = UploadSubmission.objects.filter(
+            dot_fec__report__committee_account__committee_id=committee_id
+        ).count()
+        web_print_submission_count = WebPrintSubmission.objects.filter(
+            dot_fec__report__committee_account__committee_id=committee_id
+        ).count()
+
+        self.stdout.write(self.style.SUCCESS(f"Deleting Reports: {report_count}"))
+        self.stdout.write(self.style.SUCCESS(
+            f"Deleting Transactions: {transaction_count}"
+        ))
+        self.stdout.write(self.style.SUCCESS(f"Memos: {memo_count}"))
+        self.stdout.write(self.style.SUCCESS(f"Dot Fecs: {dot_fec_count}"))
+        self.stdout.write(self.style.SUCCESS(
+            f"Upload Submissions: {upload_submission_count}"
+        ))
+        self.stdout.write(self.style.SUCCESS(
+            f"WebPrint Submissions: {web_print_submission_count}"
+        ))
+
+        reports.delete()
+        transactions.hard_delete()
```

Comment:
@Elaine-Krauss-TCG  I think we can refactor this and the reports.views.e2e_delete_all_reports() method and reduce the code duplication.

Reply:
There are a few obstacles to using that code directly.  First, the function itself is inside the Reports Viewset.  It would be awkward to instantiate the Viewset just to use the e2e_delete_all_reports method, but it could be done.  The other obstacle is the logging.  Management commands use stdout to put messages in the console as opposed to the structlog logger.  There are ways around this, creating two separate wrapper methods that handle the logging in different ways for example, but they have their own inconveniences - passing the six different count values out.  None of these are difficult challenges, not at all, but at the time it seemed more inelegant than just having two functions.  I would be happy to revisit this if it would be helpful, though.